### PR TITLE
Added recipe for Flask-Principal

### PR DESCRIPTION
**Pull request by pmlandwehr**

### CODE REVIEW

File: recipes/flask-principal/meta.yaml [hunk 1]
```diff
@@ -0,0 +1,45 @@
...
+    - flask
+
+test:
+  imports:
+    - flask_principal
+
+about:
+  home: http://packages.python.org/Flask-Principal/
+  license: MIT
+  # license_file: MANIFEST w/ LICENSE not in current PyPI release, so ommitted
```
Comment (jakirkham) [inline recipes/flask-principal/meta.yaml]: Can we please raise an issue upstream?

File: recipes/flask-principal/meta.yaml [hunk 2]
```diff
@@ -0,0 +1,45 @@
...
+
+build:
+  number: 0
+  script: python setup.py install --single-version-externally-managed --record=record.txt
+
+requirements:
+  build:
+    - python
+    - setuptools
+    - blinker
```
Comment (jakirkham) [inline recipes/flask-principal/meta.yaml]: Do we need this at run? Is it actually needed for the build?

File: recipes/flask-principal/meta.yaml [hunk 3]
```diff
@@ -0,0 +1,45 @@
+{%set name = "flask-principal" %}
+{%set camelName = "Flask-Principal" %}
```
Comment (jakirkham) [inline recipes/flask-principal/meta.yaml]: [Ditto]( https://github.com/conda-forge/staged-recipes/pull/1374#discussion_r76538237 ).

File: recipes/flask-principal/meta.yaml (hunk 1 above)
Comment (pmlandwehr) [inline recipes/flask-principal/meta.yaml]: Just didn't see the MANIFEST.in I guess. It's there, all is well.

**Comment by pmlandwehr** [comment]
Turns out the [`MANIFEST.in`](https://github.com/mattupstate/flask-principal/blob/master/MANIFEST.in) isn't in the pypi version, just the github version. Patched.

File: recipes/flask-principal/meta.yaml [hunk 4]
```diff
@@ -0,0 +1,47 @@
...
+    - flask
+
+test:
+  imports:
+    - flask_principal
+
+about:
+  home: http://packages.python.org/Flask-Principal/
+  license: MIT
+  license_file: LICENSE
```
Comment (jakirkham) [inline recipes/flask-principal/meta.yaml]: No worries. Let's just comment this and include a link to PR ( https://github.com/mattupstate/flask-principal/pull/16 ) in the comment.

File: recipes/flask-principal/14346d1.patch [hunk 5]
```diff
@@ -0,0 +1,7 @@
+diff --git MANIFEST.in MANIFEST.in
+new file mode 100644
+index 0000000..1aba38f
+--- /dev/null
++++ b/MANIFEST.in
+@@ -0,0 +1 @@
++include LICENSE
```
Comment (jakirkham) [inline recipes/flask-principal/14346d1.patch]: If only... 😩 They need this file to generate the `sdist` though.

Let's go ahead drop this. Sorry for the headache.

File: recipes/flask-principal/meta.yaml [hunk 6]
```diff
@@ -0,0 +1,47 @@
...
+  name: {{ name|lower }}
+  version: {{ version }}
+
+source:
+  fn: {{ name }}-{{ version }}.{{ compress_type }}
+  url: https://pypi.io/packages/source/{{ name[0] }}/{{ name }}/{{ name }}-{{ version }}.{{ compress_type }}
+  {{ hash_type }}: {{ hash_val }}
+  patches:
+    - 14346d1.patch
+    # Add MANIFEST.in with LICENSE - see https://github.com/mattupstate/flask-principal/blob/master/MANIFEST.in
```
Comment (jakirkham) [inline recipes/flask-principal/meta.yaml]: As mentioned above, let's drop this.

**Comment by pmlandwehr** [comment]
(ping @ocefpaf for merge)

**Comment by jakirkham** [comment]
Thanks @pmlandwehr.

**Pull request merged**